.github/workflows/sonarcloud.yml

```
@@ -0,0 +1,62 @@
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Automatically provided by GitHub
+        SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}     # You need to add this secret
+      run: |
+        cd backend
+        mvn sonar:sonar \
+          -Dsonar.projectKey=sandeepkamathkr_monthlyexpense \
+          -Dsonar.organization=sandeepkamathkr \
+          -Dsonar.host.url=https://sonarcloud.io \
+          -Dsonar.coverage.jacoco.xmlReportPaths=target/site/jacoco/jacoco.xml
+        
+    - name: Check SonarCloud Quality Gate
+      uses: sonarqube-quality-gate-action@master
```

actionlint (1.7.7)

specifying action "sonarqube-quality-gate-action@master" in invalid format because owner is missing. available formats are "{owner}/{repo}@{ref}" or "{owner}/{repo}/{path}@{ref}"

(action)